container-core/src/main/java/com/yahoo/processing/response/DataList.java

```diff
@@ -73,6 +72,10 @@
      * Making this call on a list which does not support future data always returns immediately and
      * causes no memory synchronization cost.
      */
+    CompletableFuture<DataList<DATATYPE>> future();
```

Comment:
Isn't "future" here (and the other places) a bit meaningless? It's a future where ...?

Reply:
How about completed()? (Or completeFuture())

Reply:
Good point - I agree. Your second proposal matches well with the existing method. I'll rename.